### ISSUE TITLE
Feature: Disable cache from pyproject.toml

### ISSUE THREAD

**Issue by orlandodiaz** [astral-sh/ruff#15151]
We have can pass --no-cache via the CLI to disable cache. Do we not have a setting to do this in the pyproject.toml?

A simple setting like this would be nice. Ruff is fast enough for me that cache only brings problems and adds a bunch of .ruff_cache folders everywhere. I prefer to keep it disabled

[tool.ruff]
cache = false

**Comment by MichaReiser** [comment]
There is no built-in way to disable the cache. However, on Unix, you can set `cache-dir = "/dev/null".  You can also set `cache-dir="some/shared/directory/.ruff-cache` to use a shared directory between project, to reduce the "cache folders everywhere"

> cache only brings problems and

Can you tell me more about the problems you run into with caching? It would be great if we can fix those.

**Comment by bkurtz** [comment]
Can't speak for the OP, but I like to run CI checks in a read-only copy of the working tree (helps ensure decent code hygiene like not writing files in the code tree, etc) and between that and docker, arranging for there to be a _writable_ cache dir can be a bit of a PITA.

**Comment by MichaReiser** [comment]
You can set the `RUFF_NO_CACHE` env variable in CI to disable caching.

**Comment by bkurtz** [comment]
Yup, that (or even the CLI flag) is definitely good enough for my use-case.  (I actually also really like your suggestion of setting the cache dir somewhere outside the repo, which is what I did in the end even for CI.)
